src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -945,10 +946,13 @@ public void removeWorkOrderInView(final int orderId)
      * @param building The upgraded building.
      * @param level    The new level.
      */
-    public void onBuildingUpgradeComplete(@NotNull final AbstractBuilding building, final int level)
+    public void onBuildingUpgradeComplete(@Nullable final AbstractBuilding building, final int level)
```

Comment:
May i ask why this is nullable? shouldn't a "buildingUpgrade" have a building being upgraded? (Just asking)

Reply:
remove building can be on a null building. It's basically a 0 upgrade

Reply:
ahhhh, makes sense